### PR TITLE
libtorch 1.8.0 torchvision 0.9.0

### PR DESCRIPTION
**Pull request by bayandin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by carlocab** [comment]
An earlier attempt: #72522

**Comment by carlocab** [comment]
They have `USE_NUMPY=ON` by default: https://github.com/pytorch/pytorch/blob/37c1f4a7fef115d719104e871d0cf39434aa9d56/CMakeLists.txt#L212

Makes me think this should probably depend on `numpy`...

**Comment by bayandin** [comment]
> They have `USE_NUMPY=ON` by default: https://github.com/pytorch/pytorch/blob/37c1f4a7fef115d719104e871d0cf39434aa9d56/CMakeLists.txt#L212
> 
> Makes me think this should probably depend on `numpy`...

We have `-DBUILD_PYTHON=OFF`, so I believe we don't need `numpy` for a non-python build (also, there were no problems with the previous `libtorch` build).

**Comment by carlocab** [comment]
>We have -DBUILD_PYTHON=OFF, so I believe we don't need numpy for a non-python build

I thought that meant it wouldn't build its own Python... though, I'm also confused about Python being a build-only dependency. It creates a virtual env in its prefix, and installs some Python resources into them. Don't those break without an existing Python install?

**Comment by bayandin** [comment]
> I thought that meant it wouldn't build its own Python...

I skimmed through `pytorch` CMake files and they definitely don't build their own python.

> It creates a virtual env in its prefix, and installs some Python resources into them. 

This could be done by mistake and it should be `buildpath` instead of `libexec`. It looks like python is required for some build scripts.

Currently we don't have anything python related (apart of python venv) in installation:
```
/usr/local/Cellar/libtorch/1.7.1_1$ ls libexec/
bin/        include/    lib/        pyvenv.cfg  

/usr/local/Cellar/libtorch/1.7.1_1$ ls libexec/lib/python3.9/site-packages/
PyYAML-5.3.1-py3.9.egg-info   distutils-precedence.pth      pip-21.0.1.dist-info          setuptools-53.0.0.dist-info   yaml
_distutils_hack               homebrew_deps.pth             pkg_resources                 typing-3.7.4.3-py3.9.egg-info
_yaml.cpython-39-darwin.so    pip                           setuptools                    typing.py
```

File: Formula/libtorch.rb [hunk 1]
```diff
@@ -51,11 +55,18 @@ def install
...
+      -DUSE_OPENMP=OFF
       -DUSE_SYSTEM_EIGEN_INSTALL=ON
     ]
 
     mkdir "build" do
       system "cmake", "..", *std_cmake_args, *args
+
+      # Avoid references to Homebrew shims
+      inreplace "caffe2/core/macros.h",
+                "{\"CXX_COMPILER\", \"/usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++\"},",
```
Comment (carlocab) [inline Formula/libtorch.rb:67]:
```suggestion
                "{\"CXX_COMPILER\", \"#{HOMEBREW_SHIMS_PATH}/mac/super/clang++\"},",
```

**Review by carlocab — APPROVED** [review]
Still no ARM support, but I guess that can wait for another version bump.

**Pull request closed without merging**